tests/profile.py

```diff
@@ -0,0 +1,584 @@
+        choices=[False, True],
+        type=cast_to_bool,
+        default=False,
+        help=("If True, will skip any failures upon file reads. Otherwise an exception will halt testing")
+    )
+
+    # ------------------------------------------------
+    # Main parser
+    # ------------------------------------------------
+    parser = argparse.ArgumentParser(
+        description=("Profile the performance of your file system by processing a set of files. "
+                     "A set of files is dictated by either providing a conductor job id, or a "
+                     "list of directories. Use the appropriate sub command to use either methodology.")
+    )
+    subparsers = parser.add_subparsers(title="actions")
+
+    # ------------------------------------------------
+    # Job/jid parser
+    # ------------------------------------------------
+    job_parser = subparsers.add_parser(
+        "job",
+        parents=[common_parser],
+        description=("Profile the performance of your file system by processing files from the provided "
+                     "Conductor Job ID (jid)"),
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+
+    job_parser.add_argument(
+        "jid",
+        help="The jid (job id) for a job whose files to target",
+    )
+
+    job_parser.set_defaults(func=run_job_profiler)
+
+    # ------------------------------------------------
+    # Directory parser
+    # ------------------------------------------------
+    dir_parser = subparsers.add_parser(
+        "dirs",
+        parents=[common_parser],
+        description=("Profile the performance of your file system by processing files found in the"
+                     "the given directories"),
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter
+    )
+
+    dir_parser.add_argument(
+        "dirs",
+        nargs="+",
+        help="One or more directories to read files from (recursively)",
+    )
+
+    dir_parser.set_defaults(func=run_dir_profiler)
+
+    return parser.parse_args()
+
+
+def cast_to_bool(string):
+    '''
+    Ensure that the argument provided is either "True" or "False (or "true" or
+    "false") and convert that argument to an actual bool value (True or False).
+    '''
+    string_lower = string.lower()
+    if string_lower == "true":
+        return True
+    elif string_lower == "false":
+        return False
+    raise argparse.ArgumentTypeError('Argument must be True or False')
+
+
+def run_job_profiler(args):
+
+    # ------------------------------------------------
+    # Fetch Job
+    # ------------------------------------------------
+    logger.info("Fetching Job %s", args.jid)
+    client = api_client.ApiClient()
+    r_body, _ = client.make_request('api/v1/jobs',
+                                    params="filter=jid_eq_%s" % args.jid,
+                                    use_api_key=True
+                                    )
+
+    jobs = json.loads(r_body).get("data")
```

Comment:
`client.make_request` returns a `requests` object, you should be able to parse its payload as JSON with `r_body.get_json().get('data')`.

Reply:
This comment also applies to the other use of `json.loads` in this script.

Reply:
`client.make_request` does not return a response object (sadly).   Only text (`response.text`)